AIDevGallery/Utils/AIToolkitHelper.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using AIDevGallery.Models;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace AIDevGallery.Utils;
+
+internal static class AIToolkitHelper
+{
+    private static Dictionary<AIToolkitAction, ToolkitActionInfo> aiToolkitActionInfos = new()
+    {
+        { AIToolkitAction.FineTuning, new ToolkitActionInfo() { DisplayName = "Fine Tuning", QueryName = "open_fine_tuning" } },
+        { AIToolkitAction.PromptBuilder, new ToolkitActionInfo() { DisplayName = "Prompt Builder", QueryName = "open_prompt_builder" } },
+        { AIToolkitAction.BulkRun, new ToolkitActionInfo() { DisplayName = "Bulk Run", QueryName = "open_bulk_run" } },
+        { AIToolkitAction.Playground, new ToolkitActionInfo() { DisplayName = "Playground", QueryName = "open_playground" } }
+    };
+
+    public static Dictionary<AIToolkitAction, ToolkitActionInfo> AIToolkitActionInfos
+    {
+        get { return aiToolkitActionInfos; }
+    }
+
+    public static string CreateAiToolkitDeeplink(AIToolkitAction action, ModelDetails modelDetails)
+    {
+        ToolkitActionInfo? actionInfo;
+        string deeplink = "vscode://ms-windows-ai-studio.windows-ai-studio/";
+        string modelId = action == AIToolkitAction.FineTuning ? modelDetails.AIToolkitFinetuningId! : modelDetails.AIToolkitId!;
+
+        if(aiToolkitActionInfos.TryGetValue(action, out actionInfo) && !string.IsNullOrEmpty(modelId))
+        {
+            deeplink = deeplink + $"{actionInfo.QueryName}?model_id={modelId}&track_from=AIDevGallery";
+        }
+
+        return deeplink;
+    }
+
+    public static bool ValidateForFineTuning(this ModelDetails modelDetails)
```

Comment:
Since these are extension methods, I would make sure that all places calling them are using modelDetails.ValidateForFineTuning(). Same for all methods.